test/evp_test.c

```diff
@@ -2413,6 +2413,23 @@ static char *take_value(PAIR *pp)
     return p;
 }
 
+static int key_disabled(EVP_PKEY *pkey)
+{
+#if defined(OPENSSL_NO_SM2) && !defined(OPENSSL_NO_EC)
```

Comment:
I think this line could do with a little bit more bang (!)

Reply:
No - this is correct. If we have asked for sm2 to be disabled, but ec is still enabled then we may end up reading ec keys with an sm2 group. We need to detect those and ignore them.

Reply:
Would an easier solution be to not add the NID_sm2 to the `curve_list[]` in ec_curve.c?